dmipy/core/modeling_framework.py

```diff
@@ -2191,6 +2193,383 @@ def __call__(self, acquisition_scheme, **kwargs):
+                 quantity="signal", **kwargs):
+        """
+        The MultiCompartmentModel function call for to generate signal
+        attenuation for a given acquisition scheme and model parameters.
+
+        First, the linked parameters are added to the optimized parameters.
+
+        Then, every model in the MultiCompartmentModel is called with the right
+        parameters to recover the part of the signal attenuation of that model.
+        The resulting values are multiplied with the volume fractions and
+        finally the combined signal attenuation is returned.
+
+        Aside from the signal, the function call can also return the Fiber
+        Orientation Distributions (FODs) when a dispersed model is used, and
+        can also return the stochastic cost function for the MIX algorithm.
+
+        Parameters
+        ----------
+        acquisition_scheme : DmipyAcquisitionScheme instance,
+            An acquisition scheme that has been instantiated using dMipy.
+        quantity : string
+            can be 'signal' or 'FOD' depending on the need of the model.
+        kwargs: keyword arguments to the model parameter values,
+            Is internally given as **parameter_dictionary.
+        """
+        if quantity == "signal" or quantity == "FOD":
+            values = 0
+        elif quantity == "stochastic cost function":
+            values = np.empty((
+                acquisition_scheme_or_vertices.number_of_measurements,
+                len(self.models)
+            ))
+            counter = 0
+
+        kwargs = self.add_linked_parameters_to_parameters(
+            kwargs
+        )
+        if len(self.models) > 1:
+            partial_volumes = [
+                kwargs[p] for p in self.partial_volume_names
+            ]
+        else:
+            partial_volumes = [1.]
+
+        for model_name, model, partial_volume in zip(
+            self.model_names, self.models, partial_volumes
+        ):
+            parameters = {}
+            for parameter in model.parameter_ranges:
+                parameter_name = self._inverted_parameter_map[
+                    (model, parameter)
+                ]
+                parameters[parameter] = kwargs.get(
+                    # , self.parameter_defaults.get(parameter_name)
+                    parameter_name
+                )
+
+            if quantity == "signal":
+                values = (
+                    values +
+                    partial_volume * model(
+                        acquisition_scheme_or_vertices, **parameters)
+                )
+            elif quantity == "FOD":
+                try:
+                    values = (
+                        values +
+                        partial_volume * model.fod(
+                            acquisition_scheme_or_vertices, **parameters)
+                    )
+                except AttributeError:
+                    continue
+            elif quantity == "stochastic cost function":
```

Comment:
remove this stochastic cost function bit, it's for the MIX solver and we don't have that here.